### PR TITLE
Add more robust handling of getting local IP

### PR DESCRIPTION
**Pull request by johankj**
This will fix issue #472

### CODE REVIEW

File: tchannel/net.py [hunk 1]
```diff
@@ -43,7 +43,14 @@ def interface_ip(interface):
 
 def local_ip():
     """Get the local network IP of this machine"""
-    ip = socket.gethostbyname(socket.gethostname())
+    ip = '127.0.0.1'
+    try:
+        ip = socket.gethostbyname(socket.gethostname())
+    except socket.gaierror, e:
```
Comment (westover) [inline tchannel/net.py]: Please use `socket.gaierror:`

File: tchannel/net.py [hunk 2]
```diff
@@ -43,7 +43,14 @@ def interface_ip(interface):
...
+        try:
+            ip = socket.gethostbyname(socket.gethostname() + '.local')
+        except: pass
+    except: pass
```
Comment (westover) [inline tchannel/net.py]: Extra exception

File: tchannel/net.py [hunk 3]
```diff
@@ -43,7 +43,14 @@ def interface_ip(interface):
...
 def local_ip():
     """Get the local network IP of this machine"""
-    ip = socket.gethostbyname(socket.gethostname())
+    ip = '127.0.0.1'
+    try:
+        ip = socket.gethostbyname(socket.gethostname())
+    except socket.gaierror, e:
+        try:
+            ip = socket.gethostbyname(socket.gethostname() + '.local')
+        except: pass
```
Comment (westover) [inline tchannel/net.py]: `except socket.gaierror:`

**Comment by abhinav** [comment]
Ported to #488.

**Pull request closed without merging**